components/uniset/src/provider.rs

```diff
@@ -335,17 +335,112 @@ impl Default for UnicodePropertyV1<'static> {
 
 impl<'data> UnicodePropertyV1<'data> {
     #[allow(missing_docs)] // TODO(#1030) - Add missing docs.
-    pub fn from_uniset(set: &UnicodeSet, name: Cow<'data, str>) -> UnicodePropertyV1<'data> {
+    pub fn from_uniset(set: &'data UnicodeSet, name: Cow<'data, str>) -> UnicodePropertyV1<'data> {
         UnicodePropertyV1 {
             name,
             inv_list: set.clone(),
         }
     }
+
+    #[allow(missing_docs)] // TODO(#1030) - Add missing docs.
+    pub fn from_owned_uniset(
+        set: UnicodeSet<'data>,
+        name: Cow<'data, str>,
+    ) -> UnicodePropertyV1<'data> {
+        UnicodePropertyV1 {
+            name,
+            inv_list: set,
+        }
+    }
 }
 
-impl<'data> TryInto<UnicodeSet> for UnicodePropertyV1<'data> {
-    type Error = crate::UnicodeSetError;
-    fn try_into(self) -> Result<UnicodeSet, Self::Error> {
+impl<'data> TryInto<UnicodeSet<'data>> for UnicodePropertyV1<'data> {
+    type Error = crate::UnicodeSetError<'data>;
+    fn try_into(self) -> Result<UnicodeSet<'data>, Self::Error> {
         Ok(self.inv_list)
     }
 }
+
+#[cfg(feature = "provider_serde")]
```

Comment:
If I rely on the custom derive, I get 2 nearly identical errors -- one for `SeqAccess`, one for `MapAccess`. Here is one:

```
error[E0495]: cannot infer an appropriate lifetime for lifetime parameter `'de` due to conflicting requirements
   --> components/uniset/src/provider.rs:326:5
    |
326 |     pub inv_list: UnicodeSet<'data>,
    |     ^^^
    |
note: first, the lifetime cannot outlive the lifetime `'data` as defined on the impl at 324:30...
   --> components/uniset/src/provider.rs:324:30
    |
324 | pub struct UnicodePropertyV1<'data> {
    |                              ^^^^^
note: ...so that the types are compatible
   --> components/uniset/src/provider.rs:326:5
    |
326 |     pub inv_list: UnicodeSet<'data>,
    |     ^^^
    = note: expected `Deserialize<'_>`
               found `Deserialize<'_>`
note: but, the lifetime must be valid for the lifetime `'de` as defined on the impl at 322:30...
   --> components/uniset/src/provider.rs:322:30
    |
322 |     derive(serde::Serialize, serde::Deserialize)
    |                              ^^^^^^^^^^^^^^^^^^
note: ...so that the types are compatible
   --> components/uniset/src/provider.rs:326:5
    |
326 |     pub inv_list: UnicodeSet<'data>,
    |     ^^^
    = note: expected `SeqAccess<'_>`
               found `SeqAccess<'de>`
    = note: this error originates in the derive macro `serde::Deserialize` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
So it seems like `derive(Serialize)` still works. I _think_ this has to do with `serde(borrow)` but i'm not quite sure what's going on here

Reply:
I figured it out. Firstly, the signature on `UnicodeSet` should be `impl<'de: 'a, 'a> serde::Deserialize<'de> for UnicodeSet<'a>`. Secondly, there should be at least one `serde(borrow)` in this struct, probably one one each field.

Reply:
That works, thanks!